sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -0,0 +1,30 @@
+"""
+    cmis.py
+
+    Implementation of XcvrApi that corresponds to CMIS
+"""
+
+from ...fields import consts
+from ..xcvr_api import XcvrApi
+from ..coh_optic_api import CoherentOpticApi
+
+class CmisApi(XcvrApi):
+    NUM_CHANNELS = 8
+
+    def __init__(self, xcvr_eeprom):
+        super(CmisApi, self).__init__(xcvr_eeprom)
+
+    def get_model(self):
+        return self.xcvr_eeprom.read(consts.VENDOR_PART_NO_FIELD)
+
+    # TODO: other XcvrApi methods
+
+    def _is_coherent_optic(self):
+        # TODO: check module media interface code
+        return True
+
+    def get_coherent_optic_api(self):
+        api = None
+        if self._is_coherent_optic():
+            api = CoherentOpticApi(self.xcvr_eeprom)
```

Comment:
Removed as coherent optics have now been shifted to being outside the scope of this refactor, but coherent optics are a category of CMIS-compliant modules that are not identified by a particular xcvr ID in the way a CMIS-compliant module would. The idea was that any coherent-optic specific functionality would go in a CoherentOpticApi instead of cluttering CmisApi, and CmisApi would provide a reference to the CoherentOpticApi if the module is a CMIS-compliant coherent optic.